site/profile/manifests/device_manager.pp

```diff
@@ -0,0 +1,39 @@
+class profile::device_manager (
+  Hash $defaults = lookup('profile::device_manager::defaults'),
+  Hash $devices = lookup('profile::device_manager::devices'),
+) {
```

Comment:
Those default lookups are meaningless. APL will do exactly those lookups and the calls to do them again as default values would only take place if there were no values in the first place, and the result would then be an error as just a `lookup` (without default) will fail if there is no value.

Drop both calls.

Reply:
You may want to use empty hashes as the defaults and then either just do nothing, or warn.

Reply:
i.e. just do this:
```puppet
class profile::device_manager (
  Hash $defaults,
  Hash $devices,
) {
```
That will error if a value is not bound  in hiera, or not given - that is probably what you want. If you want more validation add more specific `Struct` datatype instead of `Hash[Any, Any]` which is what you allow now (including empty hashes).